Allow ranges to be out of packet number order (#15)
wcsmith/draft-quic-receive-ts: draft-smith-quic-receive-ts.md

Each Timestamp Range describes a series of contiguous packet receive timestamps
in descending sequential packet number (and descending timestamp) order.
Timestamp Ranges consist of a Delta Largest Acknowledged indicating the
largest packet number in the range, followed by a list of Timestamp Deltas
describing the relative receive timestamps for each contiguous packet in the

> Timestamp Range (descending). Though packets within a range are in descending
> packet number and timestamp order, the ranges do not have to be in descending
> packet number order.

Reviewer: This clarifies my earlier confusion.

```suggestion
Timestamp Range (descending). Packets within a range are in descending
packet number and timestamp order. Ranges are in descending timestamp order but do not have to be in descending packet number order.
```

Timestamp Ranges are structured as shown in {{fig-ts-range}}.

~~~
Timestamp Range {

>   Delta Largest Acknowledged (i),

Reviewer: This solves the problem of representing out of order packet numbers but it doesn't not solve representing the timestamps. The timestamp delta basis in each range cannot be the largest acknowledged timestamp because that would be negative for out of order packets.

I think what we need is similar to this but with fields for latest_received_packet number and timestamp that would act as the basis for both the packet number gaps and the timestamp delta. @sharadjaiswal and I will update https://github.com/wcsmith/draft-quic-receive-ts/pull/13 with that shortly.

Author: I think it works, but maybe I'm missing something?

Reviewer: You're right. We worked through an example and it does work.
